alws/crud/build_node.py

```diff
@@ -134,11 +144,48 @@ async def __process_rpms(pulp_client: PulpClient, task_id: int, task_arch: str,
                     href=href
                 )
             )
+    if exceptions:
+        raise ArtifactConversionError(
+            'Cannot put RPM packages into Pulp storage: %s',
+            '\n'.join([str(e) for e in exceptions]))
+    for repo_href, packages in repo_packages_mapping.items():
+        try:
+            await pulp_client.modify_repository(repo_href, add=packages)
+        except Exception as e:
+            logging.error('Cannot add RPM packages to the repository: %s',
+                          str(e))
+            exceptions.append(e)
+        try:
+            await pulp_client.create_rpm_publication(repo_href)
+        except Exception as e:
+            logging.error('Cannot create a publication for the repository: %s',
+                          str(e))
+            exceptions.append(e)
+    if exceptions:
+        raise RepositoryAddError(
+            'Cannot save RPM packages into Pulp repositories: %s',
+            '\n'.join([str(e) for e in exceptions]))
+
+    if module_index:
+        packages_info = {}
+        for rpm in rpms:
+            rpm_package = await pulp_client.get_rpm_package(rpm.href)
+            packages_info[rpm.id] = rpm_package
```

Comment:
Same as above, this can be moved to `asyncio.gather()`, which will seriously improve overall execution speed
Also, maybe we should consider to make single request for all packages using `pkgId__in` (which is actually sha256)
https://docs.pulpproject.org/pulp_rpm/restapi.html#operation/content_rpm_packages_list

Reply:
We do not have SHA256 info in this part and adding it will need modifications on build node side